src/app/CommandSender.h

```diff
@@ -114,6 +114,62 @@ class CommandSender final : public Messaging::ExchangeDelegate
         virtual void OnDone(CommandSender * apCommandSender) = 0;
     };
 
+    // CommandSender methods are called outside of SDK. By having parameters passed using this
```

Comment:
What I am trying to suggest is while we have examples and tools that allocate and send commands using `CommandSender` within the SDK, there is also code outside of the SDK that allocates an instance of `CommandSender` and call the legacy APIs. My understanding is that we cannot completely change the `PrepareCommand` API parameters without also supporting legacy APIs. While creating the new `PrepareCommand` API, I am trying to do it such that the the changing parameter itself is defined within CommandSender.h, so we can expand the struct instead of expanding the default parameter args passed into the API.

Does that make sense? Or has my assumption of usage of `CommandSender` by code outside of the SDK incorrect

Reply:
No, that all makes sense.  Maybe a better way to describe that in the comment is:
```
PrepareCommand is a public SDK API, so we cannot break source compatibility for it. [etc]
```